--- xmorph-20060130/gtkmorph/Makefile.am.orig	2006-01-29 11:01:03.000000000 +0100
+++ xmorph-20060130/gtkmorph/Makefile.am	2006-02-04 16:58:22.545322500 +0100
@@ -99,11 +99,11 @@
 if COND_GTK2
 # HACK $(GTHREAD_LIBS) 
 gtkmorph_LDADD =  $(GTK_LIBS) $(GTHREAD_LIBS) $(GDK_PIXBUF_LIBS)  $(INTLLIBS) \
- $(NON_GUI_LIBS)  ../libmorph/libmorph.la 	 \
+ $(NON_GUI_LIBS)  ../libmorph/libxmorph.la 	 \
     ../glade2/interface.o ../glade2/support.o $(wl) $(fl)
 else
 gtkmorph_LDADD = $(GTK_LIBS) $(GDK_PIXBUF_LIBS) $(INTLLIBS) \
- $(NON_GUI_LIBS)  ../libmorph/libmorph.la 	 \
+ $(NON_GUI_LIBS)  ../libmorph/libxmorph.la 	 \
     ../glade1/interface.o ../glade1/support.o $(wl) $(fl)
 endif
 
--- xmorph-20060130/libmorph/Makefile.am.orig	2006-01-28 15:50:16.000000000 +0100
+++ xmorph-20060130/libmorph/Makefile.am	2006-02-04 16:58:43.834653000 +0100
@@ -7,13 +7,13 @@
 #noinst_LIBRARIES =  libmorph.a
 
 #LIBTOOL support
-lib_LTLIBRARIES = libmorph.la
+lib_LTLIBRARIES = libxmorph.la
 
-libmorph_la_LDFLAGS = -version-info 1:0:0
+libxmorph_la_LDFLAGS = -version-info 1:0:0
 
 pkginclude_HEADERS = mesh.h mesh_t.h mesh-getext.h  RgbaImage.h    my_malloc.h  spl-array.h  spline.h  tga.h  warp.h warp2.h relax.h resample.h  warp-gtk.h  
 
-libmorph_la_SOURCES =   my_malloc.c     warp.c   \
+libxmorph_la_SOURCES =   my_malloc.c     warp.c   \
   mesh.c         spline.c   my_malloc.h  tga.c   tga.h  \
   RgbaImage.c         spl-array.c   warp2.c \
   relax.c  resample.c braindead_msvc.h sinc_256.h resample_snippet.h \
--- xmorph-20060817/morph/Makefile.am.orig	2024-02-03 18:48:43.628688400 +0100
+++ xmorph-20060817/morph/Makefile.am	2024-02-03 18:50:20.884828185 +0100
@@ -8,7 +8,7 @@ INCLUDES = 	-I$(top_srcdir)/libmorph -I$
 
 bin_PROGRAMS = morph
 
-morph_LDADD =   $(INTLLIBS)  ../libmorph/libmorph.la $(NON_GUI_LIBS)
+morph_LDADD =   $(INTLLIBS)  ../libmorph/libxmorph.la $(NON_GUI_LIBS)
 
 morph_SOURCES =    main.c
 
--- xmorph-20060130/xmorph/Makefile.am.orig	2006-01-28 15:10:17.000000000 +0100
+++ xmorph-20060130/xmorph/Makefile.am	2006-02-04 16:58:59.583637250 +0100
@@ -36,4 +36,4 @@
 
 
 xmorph_LDADD =  main-x.o  $(X_PRE_LIBS) $(X_LIBS) $(XLIBS) $(X_EXTRA_LIBS) \
-	 $(GIMPLIBS) $(NON_GUI_LIBS) $(INTLLIBS)  ../libmorph/libmorph.la 
+	 $(GIMPLIBS) $(NON_GUI_LIBS) $(INTLLIBS)  ../libmorph/libxmorph.la 
