--- xmorph-2001jul27/RgbaImage.c.orig	2001-02-22 18:55:18.000000000 +0100
+++ xmorph-2001jul27/RgbaImage.c	2003-05-11 19:35:46.000000000 +0200
@@ -28,6 +28,9 @@
 # ifdef OLD_GIMP
 #   include "gimp_compat.h"
 # endif /* OLD_GIMP */
+# if (GIMP_MAJOR_VERSION < 1) || (GIMP_MINOR_VERSION < 3)
+#   define drawable_id id
+# endif
 # include <string.h>
 /* Number of steps in the progress meter. */
 extern int prog_step;
@@ -572,10 +575,10 @@
   if(rgbaImageAlloc(imgP, d->width, d->height))
     return -1;
 
-  dtype = gimp_drawable_type (d->id);
+  dtype = gimp_drawable_type (d->drawable_id);
 
   /* Get the color map. */
-  cmap = gimp_image_get_cmap (gimp_drawable_image_id (d->id), &colors);
+  cmap = gimp_image_get_cmap (gimp_drawable_image_id (d->drawable_id), &colors);
 
   /* Initialize the pixel region. */
   gimp_pixel_rgn_init (&pixel_rgn, d, 0, 0, d->width, d->height, FALSE, FALSE);
@@ -673,7 +676,7 @@
   FREE (data);
 
   /* If we had no alpha channel, then set to opaque. */
-  if (!gimp_drawable_has_alpha (d->id))
+  if (!gimp_drawable_has_alpha (d->drawable_id))
     memset (imgP->ai, RGBA_IMAGE_OPAQUE, d->height * d->width);
   return 0;
 }
@@ -717,7 +720,7 @@
   guchar *data;
   guint i, j, pel, npels, bsize, tileheight, bpp;
 
-  if (d->id == -1)
+  if (d->drawable_id == -1)
     {
       fprintf (stderr, "rgbaImageGIMP: invalid drawable ID\n");
       return -1;
@@ -741,8 +744,8 @@
       return -1;
     }
 
-  bpp = gimp_drawable_bpp (d->id);
-  dtype = gimp_drawable_type (d->id);
+  bpp = gimp_drawable_bpp (d->drawable_id);
+  dtype = gimp_drawable_type (d->drawable_id);
 
   /* Initialize the pixel region. */
   gimp_pixel_rgn_init (&pixel_rgn, d, 0, 0, d->width, d->height, TRUE, FALSE);
--- xmorph-2001jul27/sequence.c.orig	2001-02-22 18:55:18.000000000 +0100
+++ xmorph-2001jul27/sequence.c	2003-05-11 19:40:34.000000000 +0200
@@ -23,6 +23,9 @@
 # ifdef OLD_GIMP
 # include "gimp_compat.h"
 # endif /* OLD_GIMP */
+# if (GIMP_MAJOR_VERSION < 1) || (GIMP_MINOR_VERSION < 3)
+#   define drawable_id id
+# endif
 
 /* Number of steps in the progress meter. */
 int prog_step = 0;
@@ -172,13 +175,13 @@
         GimpDrawable *d;
         gint32 image_ID, layer_ID;
 
-        image_ID = gimp_drawable_image_id (drawable->id);
+        image_ID = gimp_drawable_image_id (drawable->drawable_id);
 
         /* Create a new layer in the source image. */
         sprintf(iname, "%s%04i.tga", basename, frame);
         layer_ID = gimp_layer_new (image_ID, iname,
                                    dmP->src_img.ncols, dmP->src_img.nrows,
-                                   gimp_drawable_type (drawable->id),
+                                   gimp_drawable_type (drawable->drawable_id),
                                    100, GIMP_NORMAL_MODE);
         gimp_image_add_layer (image_ID, layer_ID, 0);
 
--- xmorph-2001jul27/Makefile.orig	2003-05-11 19:38:51.000000000 +0200
+++ xmorph-2001jul27/Makefile	2003-05-11 19:41:47.000000000 +0200
@@ -295,7 +295,7 @@
 # Comment out this variable if you have not used -DGIMP.
 
 #GIMPLIBS=-L/usr/local/lib -lgimp -lglib
-GIMPLIBS=-lgimp -lglib
+GIMPLIBS=-lgimp
 
 
 
@@ -391,7 +391,7 @@
 
 #CFLAGS=$(DEBUG) $(DEFINES) $(INCLUDE) -DSUNOS
 
-CFLAGS=$(OPT) $(DEFINES) $(INCLUDE) -I/usr/X11R6/include `glib-config --cflags`
+CFLAGS=$(OPT) $(DEFINES) $(INCLUDE) -I/usr/X11R6/include
 
 
 
--- xmorph-2001jul27/xmorph.c.orig	2001-02-22 18:55:18.000000000 +0100
+++ xmorph-2001jul27/xmorph.c	2003-05-11 19:43:18.000000000 +0200
@@ -651,7 +651,11 @@
      int *nreturn_vals, GimpParam ** return_vals)
 {
   static GimpParam values[2];
+#if (GIMP_MAJOR_VERSION >= 1) && (GIMP_MINOR_VERSION >= 3)
+  GimpRunMode run_mode;
+#else
   GimpRunModeType run_mode;
+#endif
   XtAppContext app;
 
   GimpPDBStatusType  status = GIMP_PDB_SUCCESS;
--- xmorph-2001jul27/file_menu.c.orig	2001-02-22 18:55:18.000000000 +0100
+++ xmorph-2001jul27/file_menu.c	2003-05-11 22:11:37.000000000 +0200
@@ -231,10 +231,10 @@
   gint32 *images;
   gint32 *layers;
   gint32 *channels;
-  int nimages;
-  int nlayers;
-  int nchannels;
-  int i, j, k;
+  int nimages = 0;
+  int nlayers = 0;
+  int nchannels = 0;
+  int i, j, k, freename;
 
   /* FIXME: how can we dynamically update this menu? */
 
@@ -251,9 +251,10 @@
     if (1)
       {
         name = gimp_image_get_filename (images[i]);
+        if(name) freename=1; else { name = "NONAME"; freename=0; }
         image_label = g_new (char, strlen (name) + 16);
         sprintf (image_label, "%s-%d", my_base_name (name), images[i]);
-        g_free (name);
+        if(freename) g_free (name);
 
         layers = gimp_image_get_layers (images[i], &nlayers);
         for (j = 0; j < nlayers; j++)
@@ -261,9 +262,10 @@
           if (1)
             {
               name = gimp_layer_get_name (layers[j]);
+              if(name) freename=1; else { name = "NONAME"; freename=0; }
               label = g_new (char, strlen (image_label) + strlen (name) + 2);
               sprintf (label, "%s/%s", image_label, name);
-              g_free (name);
+              if(freename) g_free (name);
 
               menuitem = XtVaCreateManagedWidget (label, smeBSBObjectClass,
                                                dest_menu, NULL);
@@ -280,9 +282,10 @@
           if (1)
             {
               name = gimp_channel_get_name (channels[j]);
+              if(name) freename=1; else { name = "NONAME"; freename=0; }
               label = g_new (char, strlen (image_label) + strlen (name) + 2);
               sprintf (label, "%s/%s", image_label, name);
-              g_free (name);
+              if(freename) g_free (name);
 
               menuitem = XtVaCreateManagedWidget (label, smeBSBObjectClass,
                                                dest_menu, NULL);
--- xmorph-2001jul27/main.c.orig	2001-07-27 19:02:17.000000000 +0200
+++ xmorph-2001jul27/main.c	2003-06-19 16:17:33.000000000 +0200
@@ -44,6 +44,15 @@
 
 #include "main.h"
 
+#ifdef NEED_GIMP
+# include <libgimp/gimp.h>
+# ifdef OLD_GIMP
+#   include "gimp_compat.h"
+# endif /* OLD_GIMP */
+/* from xmorph.c */
+extern GimpPlugInInfo PLUG_IN_INFO;
+#endif /* NEED_GIMP */
+
 
 
 
@@ -93,9 +102,6 @@
   return (atan((x - 0.5) * SHARPNESS * 2.0) + as) / (2.0 * as);
 }
 
-
-
-
 int
 main(int argc, char **argv)
 {
@@ -145,7 +151,11 @@
 #  endif    
 
     /* gimp_main () returns 1 if xmorph wasn't invoked by GIMP */
+#  if ((GIMP_MAJOR_VERSION > 1) || ((GIMP_MAJOR_VERSION == 1) && (GIMP_MINOR_VERSION > 3)) || ((GIMP_MAJOR_VERSION == 1) && (GIMP_MINOR_VERSION == 3) && (GIMP_MICRO_VERSION >= 15)))
+    result = gimp_main (&PLUG_IN_INFO, argc, argv);
+#  else
     result = gimp_main (argc, argv);
+#endif
 
 #  ifdef OLD_GIMP
     g_set_message_handler (old_print_func);
