--- xmorph-2001feb22/my_malloc.h.glibc	Tue Feb 20 11:22:00 2001
+++ xmorph-2001feb22/my_malloc.h	Fri May 25 20:05:08 2001
@@ -58,6 +58,7 @@
 #endif
 
 #include <stdlib.h>
+#include <string.h>
 
 
 
@@ -66,10 +67,10 @@
 // Note that if optimization is turned on, GNU C might define strdup
 // as a macro, in which case this prototype will cause bad things
 // to happen.
-*/
 #if ( defined(__GNUC__) && defined (STRDUP_PROTO_BROKEN) )
 extern char *strdup (const char *s);
 #endif
+*/
 
 
 
