--- xmorph-2001feb22/Makefile.make	Thu Feb 22 13:13:45 2001
+++ xmorph-2001feb22/Makefile	Fri May 25 19:37:36 2001
@@ -121,7 +121,7 @@
 #CC=gcc
 #CC=/bin/cc
 #CC=cc
-CC=cc
+CC=gcc
 
 
 
@@ -275,10 +275,10 @@
 
 # Linux:
 #XLIBS=-L/usr/X11/lib -lXaw -lXmu -lXext -lXt -lX11
-#XLIBS=-L/usr/X11R6/lib -lXaw -lXmu -lXext -lXt -lX11
+XLIBS=-L/usr/X11R6/lib -lXaw -lXmu -lXext -lXt -lX11
 
 # SGI IRIX:
-XLIBS=              -lXaw -lXmu -lXext -lXt -lcvt -lX11
+#XLIBS=              -lXaw -lXmu -lXext -lXt -lcvt -lX11
 
 
 
@@ -292,7 +292,7 @@
 # Comment out this variable if you have not used -DGIMP.
 
 #GIMPLIBS=-L/usr/local/lib -lgimp -lglib
-#GIMPLIBS=-lgimp -lglib
+GIMPLIBS=-lgimp -lglib
 
 
 
@@ -326,7 +326,7 @@
 #NON_GUI_LIBS = -lm
 #NON_GUI_LIBS = -lm -ldl
 #NON_GUI_LIBS = -lm /usr/lib/debug/malloc.o
-NON_GUI_LIBS = -lm -lmalloc
+NON_GUI_LIBS = -lm -ldl
 
 
 
@@ -388,14 +388,14 @@
 
 #CFLAGS=$(DEBUG) $(DEFINES) $(INCLUDE) -DSUNOS
 
-CFLAGS=$(DEBUG) $(DEFINES) $(INCLUDE)
+CFLAGS=$(OPT) $(DEFINES) $(INCLUDE) -I/usr/X11R6/include `glib-config --cflags`
 
 
 
 
 ## INSTALL_BASE: Directory base to install files
 ## ---------------------------------------------
-INSTALL_BASE=/usr/local
+INSTALL_BASE=/usr
 
 
 
@@ -624,10 +624,6 @@
 send: dist
 	rcp xmorph-$(BACKDIR).tar.gz ucsub:public_html/Software/Graphics
 
-ctags:
-	-ctags    *.[ch]
-	-ctags -t *.[ch]
-
 unproto:
 	unprotoize $(SRCS)
 
@@ -643,7 +639,7 @@
 cleandepend: ctags
 	makedepend -Y *.c
 
-depend: ctags
+depend: 
 	makedepend $(INCLUDE) $(SRCS)
 
 
